### PR TITLE
[pysearch] Get rid of Storage.

### PR DESCRIPTION
**Pull request by ygorshenin**
@bykoianko @tatiana-kondakova PTAL

### CODE REVIEW

File: search/pysearch/bindings.cpp [hunk 1]
```diff
@@ -152,13 +141,13 @@ struct SearchEngineProxy
...
-    CHECK(g_storage.get() != nullptr, ("init() was not called."));
+    CHECK(g_affiliations.get(), ("init() was not called."));
     auto & platform = GetPlatform();
     auto infoGetter = storage::CountryInfoReader::CreateCountryInfoReader(platform);
-    infoGetter->InitAffiliationsInfo(&g_storage->GetAffiliations());
+    infoGetter->InitAffiliationsInfo(&*g_affiliations);
 
     m_engine = make_shared<search::tests_support::TestSearchEngine>(
-        move(infoGetter), make_unique<search::ProcessorFactory>(), search::Engine::Params{});
+        move(infoGetter), my::make_unique<search::ProcessorFactory>(), search::Engine::Params{});
```
Comment (tatiana-kondakova) [inline search/pysearch/bindings.cpp]: `#include <utility>`

File: storage/country.hpp [hunk 2]
```diff
@@ -81,9 +83,13 @@ using TCountryTree = CountryTree<TCountryId, Country>;
...
+                                TMappingAffiliations & affiliations,
+                                TMappingOldMwm * mapping = nullptr);
+int64_t LoadCountriesFromFile(std::string const & path, TCountryTree & countries,
+                              TMappingAffiliations & affiliations,
+                              TMappingOldMwm * mapping = nullptr);
 
-void LoadCountryFile2CountryInfo(string const & jsonBuffer, map<string, CountryInfo> & id2info,
-                                 bool & isSingleMwm);
+void LoadCountryFile2CountryInfo(std::string const & jsonBuffer,
+                                 map<std::string, CountryInfo> & id2info, bool & isSingleMwm);
```
Comment (tatiana-kondakova) [inline storage/country.hpp]: std::map

File: search/pysearch/bindings.cpp [hunk 3]
```diff
@@ -66,8 +53,10 @@ void Init(string const & resource_path, string const & mwm_path)
 
   classificator::Load();
 
-  g_storage = make_unique<storage::Storage>(countriesFile, mwm_path);
-  g_storage->Init(&DidDownload, &WillDelete);
+  g_affiliations = my::make_unique<storage::TMappingAffiliations>();
+  storage::TCountryTree countries;
+  auto const rv = storage::LoadCountriesFromFile(countriesFile, countries, *g_affiliations);
+  CHECK(rv != -1, ("Can't load countries from:", countriesFile));
```
Comment (tatiana-kondakova) [inline search/pysearch/bindings.cpp]: в country.hpp к LoadCountriesFromFile можно тоже добавить коммент `/// @return version of country file or -1 if error was encountered`

Comment (ygorshenin) [reply search/pysearch/bindings.cpp]: Там один коммент на две функции.

File: search/pysearch/bindings.cpp (hunk 1 above)
Comment (ygorshenin) [reply search/pysearch/bindings.cpp]: Done

File: storage/country.hpp (hunk 2 above)
Comment (ygorshenin) [reply storage/country.hpp]: Done

**Comment by ygorshenin** [comment]
@bykoianko PTAL

File: storage/country.hpp [hunk 4]
```diff
@@ -81,9 +83,13 @@ using TCountryTree = CountryTree<TCountryId, Country>;
 using TCountryTreeNode = TCountryTree::Node;
```
Comment (bykoianko) [inline storage/country.hpp]: Let's move the using up to the other usings

File: storage/country.cpp [hunk 5]
```diff
@@ -3,13 +3,16 @@
...
+
 #include "base/logging.hpp"
 #include "base/stl_helpers.hpp"
 
 #include "3party/jansson/myjansson.hpp"
 
-#include "std/utility.hpp"
+#include <utility>
 
+using namespace std;
```
Comment (bykoianko) [inline storage/country.cpp]: If you put `using namespace std;` here let's remove `std::` from the file. As far as I see there's some `std::` in Storage::OnMapFileDownloadFinished().

Comment (bykoianko) [reply storage/country.cpp]: Oh! Sorry.

File: storage/country.hpp (hunk 4 above)
Comment (ygorshenin) [reply storage/country.hpp]: Done

Comment (ygorshenin) [reply storage/country.hpp]: Ah, no. Unfortunately, I can't. This using declaration depends on the using declaration above, and the latter one depends on the Country.

**Comment by ygorshenin** [comment]
@bykoianko or @tatiana-kondakova , please, merge.

File: storage/country.hpp (hunk 4 above)
Comment (bykoianko) [reply storage/country.hpp]: Ok

**Pull request merged**